internal/pkg/service/stream/storage/level/local/diskreader/volume.go

```diff
@@ -127,26 +129,26 @@ func (v *Volume) Metadata() volume.Metadata {
 	}
 }
 
-func (v *Volume) OpenReader(slice *model.Slice) (r Reader, err error) {
+func (v *Volume) OpenReader(sliceKey model.SliceKey, slice localModel.Slice, encodingCompression compression.Config, stagingCompression compression.Config) (r Reader, err error) {
 	// Check context
 	if err := v.ctx.Err(); err != nil {
-		return nil, errors.PrefixErrorf(err, `reader for slice "%s" cannot be created: volume is closed`, slice.SliceKey.String())
+		return nil, errors.PrefixErrorf(err, `reader for slice "%s" cannot be created: volume is closed`, sliceKey.String())
 	}
 
 	// Setup logger
 	logger := v.logger.With(
-		attribute.String("projectId", slice.ProjectID.String()),
-		attribute.String("branchId", slice.BranchID.String()),
-		attribute.String("sourceId", slice.SourceID.String()),
-		attribute.String("sinkId", slice.SinkID.String()),
-		attribute.String("fileId", slice.FileID.String()),
-		attribute.String("sliceId", slice.SliceID.String()),
+		attribute.String("projectId", sliceKey.ProjectID.String()),
```

Comment:
Shouldn't this be part of telemetry, so the logger inherits it? Or is it incorrect and we have to do it this way?

Reply:
@michaljurecko what do you think? Is it ok to have telemetry on `diskreader` level as there is no whole `model.File` present?